pkg/handlers/default.go

```diff
@@ -49,3 +56,68 @@ func KillContainer(pid int) registry.HandlerFunc {
 		return registry.HandlerResultErrno(unix.EPERM)
 	}
 }
+
+func FreezeContainer(pid int) registry.HandlerFunc {
+	re := regexp.MustCompile(`^(\d*):(freezer)?:(/..*)$`)
+	return func(fd libseccomp.ScmpFd, req *libseccomp.ScmpNotifReq) (result registry.HandlerResult) {
+		cgroupPath := ""
+		if cgroupFile, err := os.Open(filepath.Join("/proc", fmt.Sprintf("%d", pid), "cgroup")); err == nil {
+			defer cgroupFile.Close()
+			reader := bufio.NewReader(cgroupFile)
+			for {
+				line, err := reader.ReadString('\n')
+				if err != nil {
+					break
+				}
+				line = strings.TrimSuffix(line, "\n")
+
+				r := re.FindStringSubmatch(line)
+				if len(r) < 4 {
+					continue
+				}
+
+				cgroupPath = r[3]
+				break
+			}
+		} else {
+			log.WithFields(log.Fields{
+				"pid": pid,
+				"err": err,
+			}).Error("cannot parse cgroup")
+			return registry.HandlerResultErrno(unix.EPERM)
+		}
```

Comment:
I think this is more readable if you change the if to be: if err != nil, then do this. No else, and just proceed with the err == nil case. We reduce the indentation and we just filter the "uninteresting cases" at the beginning and continue with what is interesting here (being able to open the file and freeze the container).